packages/server/src/queue/queue.controller.ts

```diff
@@ -72,21 +92,36 @@ export class QueueController {
     if (queue === undefined) {
       throw new NotFoundException();
     }
-
     queue.notes = body.notes;
     queue.allowQuestions = body.allowQuestions;
-    await queue.save();
+    try {
+      await queue.save();
+    } catch (err) {
+      console.error(err);
+      throw new HttpException(
+        ERROR_MESSAGES.queueController.saveQueue,
+        HttpStatus.INTERNAL_SERVER_ERROR,
+      );
+    }
     return queue;
   }
 
   @Post(':queueId/clean')
   @Roles(Role.TA, Role.PROFESSOR)
   async cleanQueue(@Param('queueId') queueId: number): Promise<void> {
     // Clean up queue if necessary
-    setTimeout(async () => {
-      await this.queueCleanService.cleanQueue(queueId, true);
-      await this.queueSSEService.updateQueue(queueId);
-    });
+    try {
+      setTimeout(async () => {
```

Comment:
why the `setTimeout`?

Reply:
That was a;ready there in the codebase. I just added a try block on top of it!

Reply:
think I'm going blind... thanks